src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+            statement.setBigDecimal(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        "Creating book failed , no raws affected " + book
+                );
+            }
+            book.setId(getGenerateId(statement));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a new book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Optional<Book> book = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                book = Optional.of(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Can't find a book with id = %d ", id), e
+            );
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> list = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                list.add(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB", e);
+        }
+        return list;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        String.format(
+                                "Book with id = %d wasn't updated, no rows affected", book.getId()
+                        ));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format(
+                            "Book with id = %d wasn't updated, no rows affected", book.getId()), e
+            );
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_BY_ID_QUERY)) {
+            statement.setLong(3, id);
```

Comment:
The parameter index for `statement.setLong(3, id);` is incorrect. It should be `statement.setLong(1, id);` since there is only one placeholder in the DELETE_BY_ID_QUERY.